Dip/DipTests/ComponentScopeTests.swift

```diff
@@ -90,14 +90,14 @@ class ComponentScopeTests: XCTestCase {
 
   func testThatItReusesInstanceInObjectGraphScopeDuringResolve() {
     //given
-    container.register(.ObjectGraph) { [unowned container] in Client(server: container.resolve()) as Client }
+    container.register(.ObjectGraph) { [unowned container] in Client(server: try! container.resolve()) as Client }
     
     container.register(.ObjectGraph) { Server() as Server }.resolveDependencies { container, server in
```

Comment:
I've created separate issue #23 to discuss that. Will probably add some unit test for that.